src/mscorlib/shared/System/Gen2GcCallback.cs

```diff
@@ -0,0 +1,76 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    /// <summary>
+    /// Schedules a callback roughly every gen 2 GC (you may see a Gen 0 an Gen 1 but only once)
+    /// (We can fix this by capturing the Gen 2 count at startup and testing, but I mostly don't care)
+    /// </summary>
+    internal sealed class Gen2GcCallback : CriticalFinalizerObject
+    {
+        public Gen2GcCallback()
```

Comment:
private?

Reply:
I think it can be? This is just a straight copy of the existing class that was sitting in the PinnableBufferCache.cs file.

Reply:
Yes, make it private.  I wrote this code originally and it was a mistake not to make it private. .

Reply:
Done